### PR TITLE
Update tool-Docker-action to v2.1.0

### PR DESCRIPTION
**Pull request by nwiltsie**
This PR updates the `tool-Docker-action` Action to version 2.1.0. There are a few big changes in this version:

1. Pushing to any branch `foo` will [build this repository's image](https://github.com/uclahs-cds/tool-Docker-action/pull/14) and tag it as `branch-foo`. That tagged image will be removed when the branch is deleted.
1. By default the Action will only build [SemVer](https://semver.org/) tags (e.g. `v1.2.3-xxx`). Repositories that need to use other kinds of tags (e.g. `v1.2beta`) can pass the optional `non-semver-tags` argument; doing so will build all tags that begin with `v`.

There are a bunch of smaller changes buried in there - you can see them all at [this link](https://github.com/uclahs-cds/tool-Docker-action/compare/ebe9f1c...v2.1.0).

Based on this repository's existing tags, I've determined that it _does_ follow semantic versioning. The updated workflow therefore does _not_ pass the `non-semver-tags` argument to the Action, although it can be uncommented in the future if need be.

### CODE REVIEW

File: .github/workflows/docker-build-release.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      github.event.ref_type == 'branch' && 'branch-' || '',
+      github.event.ref
+    )
+    || github.ref == 'refs/heads/main' && 'Update `dev`'
+      || format(
+        'Update `{0}{1}`',
+        !startsWith(github.ref, 'refs/tags') && 'branch-' || '',
+        github.ref_name
+      )
+  }} docker tag
```
Comment (zhuchcn) [inline .github/workflows/docker-build-release.yaml:17]: What does `run-name` do？

**Review by zhuchcn — COMMENTED** [review]
Can we keep `workflow_dispatch`? Also, is `github-token` no longer needed?

**Comment by nwiltsie** [comment]
> Can we keep `workflow_dispatch`?

Shoot, I added back in the `unstable` tagging logic but forgot to add `workflow_dispatch`. I'll fix that up.

> Also, is `github-token` no longer needed?

There is [still a `github-token` argument](https://github.com/uclahs-cds/tool-Docker-action/blob/28a9544999889b1131f77ecbc036a9ee571a578d/action.yml#L13-L15), but it defaults to `${{ github.token }}` so we generally don't need to provide it explicitly.

File: .github/workflows/docker-build-release.yaml (hunk 1 above)
Comment (nwiltsie) [reply .github/workflows/docker-build-release.yaml:17]: Screenshot 2024-08-16 at 2 23 39 PM

It controls the run's name in the "Actions" view - the screenshot above is an example. You can find some of my complaints about how irritating it is that the `run-name` must be so complex in the [original pull request](https://github.com/uclahs-cds/tool-Docker-action/pull/14).

**Comment by nwiltsie** [comment]
Okay, I've added `workflow_dispatch` back in with 8e7a9946d48aa12e165f91ee6033ab43034d8815.

File: .github/workflows/docker-build-release.yaml (hunk 1 above)
Comment (zhuchcn) [reply .github/workflows/docker-build-release.yaml:17]: Ah that's cool!

**Review by zhuchcn — APPROVED** [review]
Wonderful! Thank you!

**Pull request merged**